### ISSUE TITLE
Slack returner 'slack.changes' option doesn't appear to function

### ISSUE THREAD

**Issue by jbouse** [saltstack/salt#35507]
### Description of Issue/Question

Attempting to use the slack.changes option to reduce the output sent to Slack channel when using the returner. As documented it should only return entries that either had changes or failed to execute. My testing has not shown that to be true.

### Setup
Setup /etc/salt/minion.d/returners.conf with the follwing:
```
slack.api_key: XXX-REDACTED-XXX
slack.as_user: true
slack.channel: '#general'
slack.username: saltstack
slack.changes: true
```
Then restart the salt-minion to load the configuration

### Steps to Reproduce Issue
I have a simple route53 state that calls boto_route53.present but as I currenly have boto 2.34.0 and boto_route53 requires 2.35.0 or higher it is assured to generate a failed entry when ran.

Running **salt MINION state.sls route53 test=False --return slack --output yaml** I get the following
output on my terminal I executed it from:

```
MINION:
  boto_route53_|-r53_a_record_|-MINION_|-present:
    __run_num__: 1
    changes: {}
    comment: 'State ''boto_route53.present'' was not found in SLS ''route53''

      Reason: ''boto_route53'' __virtual__ returned False

'
    name: MINION
    result: false
  pkg_|-boto_pkg_|-python-boto_|-installed:
    __id__: boto_pkg
    __run_num__: 0
    changes: {}
    comment: Package python-boto is already installed
    duration: 413.451
    name: python-boto
    result: true
    start_time: '03:57:57.568486'
```

While this is the output I expect it to generate... I don't recieve anything in my Slack channel. If I remove the slack.changes option from the returners.conf, restart the salt-minion and then re-execute the same command I do get the report to my Slack channel for both return ID items.

### Versions Report
```
Salt Version:
           Salt: 2016.3.2

Dependency Versions:
           cffi: 0.8.6
       cherrypy: 3.5.0
       dateutil: 2.2
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
         Jinja2: 2.7.3
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.2
   mysql-python: 1.2.3
      pycparser: 2.10
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.9 (default, Mar  1 2015, 12:57:24)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.4.0
           RAET: Not Installed
          smmap: 0.8.2
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: debian 8.5
        machine: x86_64
        release: 3.2.0-4-amd64
         system: Linux
        version: debian 8.5
```

**Comment by Ch3LL** [comment]
Can you give this PR a try: #33337  

If that works we can see about backporting that to 2016.3

**Comment by jbouse** [comment]
@Ch3LL I'm happy to report that making the change from the PR works perfectly...

**Comment by jbouse** [comment]
If it'll help get it backported and released to 2016.3 quicker/easier, I can file a ticket through our support account.

**Comment by rallytime** [comment]
@jbouse I have back-ported #33337 to the 2016.3 branch with PR #35573. :)

**Comment by rallytime** [comment]
Back-port is merged into the 2016.3 branch now. The change will make it into the 2016.3.3 release. Thanks @jbouse.

**Comment by rickh563** [comment]
ZD-908
